storage-api/src/java/org/apache/hadoop/hive/ql/exec/vector/Decimal64ColumnVector.java

```diff
@@ -136,12 +136,12 @@ public void setElement(int outputElementNum, int inputElementNum, ColumnVector i
           decimal64ColVector.vector[inputElementNum], decimal64ColVector.scale);
       scratchHiveDecWritable.mutateEnforcePrecisionScale(precision, scale);
       if (scratchHiveDecWritable.isSet()) {
-        vector[inputElementNum] = scratchHiveDecWritable.serialize64(scale);
+        vector[outputElementNum] = scratchHiveDecWritable.serialize64(scale);
```

Comment:
Wong indexer? It was not covered with unit test?

Reply:
Yes, not for the primitive column vectors.